From 282a7ea2351050e96b400eebe3a0cdc47085c95a Mon Sep 17 00:00:00 2001
From: Nicolas Sanguinetti <contacto@nicolassanguinetti.info>
Date: Thu, 15 Apr 2010 09:56:45 -0300
Subject: [PATCH] Avoid deprecation warnings and potential TypeErrors

If the deprecated Base.protected_instance_variables isn't set, then you get some nasty
TypeError exceptions, so let's try to avoid that
---
 actionpack/lib/abstract_controller/assigns.rb   |    4 ++--
 actionpack/lib/abstract_controller/rendering.rb |    2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/actionpack/lib/abstract_controller/assigns.rb b/actionpack/lib/abstract_controller/assigns.rb
index 21459c6..4f58cac 100644
--- a/actionpack/lib/abstract_controller/assigns.rb
+++ b/actionpack/lib/abstract_controller/assigns.rb
@@ -6,7 +6,7 @@ module AbstractController
     def view_assigns
       hash = {}
       variables  = instance_variable_names
-      variables -= protected_instance_variables if respond_to?(:protected_instance_variables)
+      variables -= config.protected_instance_variables if config.respond_to?(:protected_instance_variables)
       variables.each { |name| hash[name] = instance_variable_get(name) }
       hash
     end
@@ -18,4 +18,4 @@ module AbstractController
       view_assigns.each { |k,v| object.instance_variable_set(k, v) }
     end
   end
-end
\ No newline at end of file
+end
diff --git a/actionpack/lib/abstract_controller/rendering.rb b/actionpack/lib/abstract_controller/rendering.rb
index d2db366..f1e6f7d 100644
--- a/actionpack/lib/abstract_controller/rendering.rb
+++ b/actionpack/lib/abstract_controller/rendering.rb
@@ -126,7 +126,7 @@ module AbstractController
     def view_assigns
       hash = {}
       variables  = instance_variable_names
-      variables -= protected_instance_variables if respond_to?(:protected_instance_variables)
+      variables -= config.protected_instance_variables if config.respond_to?(:protected_instance_variables)
       variables.each { |name| hash[name.to_s[1..-1]] = instance_variable_get(name) }
       hash
     end
-- 
1.7.0.3

